src/Proxy.php

```diff
@@ -108,10 +109,27 @@ public function isPersisted(): bool
      */
     public function object(): object
     {
-        if ($this->autoRefresh && $this->persisted) {
-            $this->refresh();
+        if (!$this->autoRefresh || !$this->persisted) {
+            return $this->object;
         }
 
+        // todo: what if object manager isn't entity manager?
```

Comment:
Do we support that in Foundry?

Reply:
Technically, yes but I've never tested with or even used an object manager other than the orm.

Reply:
I think I'll just check that it's an instance of `EntityManagerInterface` in the conditional below.